doc/influxql-2.md

```diff
@@ -0,0 +1,910 @@
+// Returns: error string, empty if none
+createDatabase(name:"foo")
+```
+
+#### Drop Database
+`dropDatabase` will drop the specified database.
+
+```javascript
+// Accepts: start
+// Arguments:
+//   name - required string
+// Returns: error string, empty if none
+dropDatabase(name:"foo")
+```
+
+#### Select
+`select` will return a matrix of series with empty vectors based on selection criteria. By default, select will limit the time range of the index it checks to the most recent data (depending on configuration this could be 24h-7d). For exact time matching use select to filter down the number of series and then perform a query that returns the count of results in that time window, then filter out all zero counts. To get any time series data back from the database, the `range` function must be chained off the select. For limiting the number of series that get returned you can use the `limit` function. You can also order the series using `sort`.
+
+**NOTE:** A major difference between 2.0 and 1.x is that by default, series will not be merged together. That is, all `select` statements will yield a matrix with each individual series. This is what `group by *` would do in InfluxQL 1.x. With InfluxQL 2.0 it is now the default behavior.
+
+```javascript
+// Accepts: start, matrix
+// Arguments:
+//   database - a string, required if not chained off a matrix
+//   where - an expression to match against tags
+// Returns: a matrix with only the series column (empty points arrays)
+select(database:"foo") // get all series for database foo
+```
+
+##### The `where` expression
+
+The `where` argument takes an expression that matches predicates against tag information. The where argument is wrapped in curley braces `{` `}`. Tag keys must be wrapped in single quotes and tag values must be wrapped in double quotes. Here are some examples:
+
+```javascript
+select(database:"foo", where:{'metric'="cpu"})
+select(database:"foo", where:{'metric'="cpu" OR 'metric'="mem"})
+select(database:"foo", where:{('metric'="cpu" OR 'metric'="mem") AND 'host'="serverA"})
+select(database:"foo", where:{'host' in ["a","b"]})
+select(database:"foo", where:{'host' NOT EMPTY})
+
+// variables can be referenced in the where expression
+var hostName = "serverA"
+select(database:"foo", where:{'host' = hostName})
+```
+
+#### Range
+`range` will get results by the time range specified. If chained against `select` it will pull the data from the database. If chained against a matrix, the points in the matrix will be filtered by the range. The range function could be updated with new arguments later to do things like filter out or keep periodic intervals (e.g. range for 4 weeks of data, but exclude Saturdays and Sundays).
+
+```javascript
+// Accepts: select or matrix
+// Arguments:
+//   start - a time. Could be relative e.g. `-4h` or an epoch, or a time string
+//   end - a time
+//   count - an integer. Get this number of values either from the passed 
```

Comment:
Yeah, I'm completely in favor of atomic primitives that can be chained for more complex functionality. We'll definitely want shortcuts for common things in the future. We could also make it something by convention. For example, function names and parameter names are always camel cased. We could use `_` to separate and combine. For example:
```
// chained
select().range(start: -4h).limit(n:10).order(by:"asc")
// combined
select(range_start: -4h, limit_n: 10, order_by: "asc")
```